src/udp/socket.rs

```diff
@@ -141,21 +143,38 @@ pub trait UdpSocketExt {
         C: 'static;
 }
 
+fn bind_reusable(addr: &SocketAddr) -> io::Result<::std::net::UdpSocket> {
+    let socket = match addr.ip() {
+        IpAddr::V4(..) => UdpBuilder::new_v4()?,
+        IpAddr::V6(..) => UdpBuilder::new_v6()?,
+    };
+    socket.reuse_address(true)?;
+
+    #[cfg(target_family = "unix")]
+    {
+        use net2::unix::UnixUdpBuilderExt;
+        socket.reuse_port(true)?;
+    }
+
+    let socket = socket.bind(addr)?;
+    Ok(socket)
+}
+
 impl UdpSocketExt for UdpSocket {
     fn bind_reusable(addr: &SocketAddr, handle: &Handle) -> io::Result<UdpSocket> {
-        let socket = match addr.ip() {
-            IpAddr::V4(..) => UdpBuilder::new_v4()?,
-            IpAddr::V6(..) => UdpBuilder::new_v6()?,
-        };
-        socket.reuse_address(true)?;
+        let socket = bind_reusable(addr)?;
+        let socket = UdpSocket::from_socket(socket, handle)?;
 
-        #[cfg(target_family = "unix")]
-        {
-            use net2::unix::UnixUdpBuilderExt;
-            socket.reuse_port(true)?;
-        }
+        Ok(socket)
```

Comment:
isn't his redundant? `UdpSocket::from_socket()` already returns `io::Result<UdpSocket>`: https://docs.rs/tokio-core/0.1.11/tokio_core/net/struct.UdpSocket.html#method.from_socket

Reply:
Yeah, but I sometimes find it clearer to write like that. I've removed it though.

Reply:
Hmm, personally for me, function signature is clear enough: `fn bind_reusable(addr: &SocketAddr, handle: &Handle) -> io::Result<UdpSocket> {`